Python pacakges tutorial

hbrincon opened the pull request:
Updated python packages tutorial that now covers numpy, matplotlib, and astropy and works with DR1 data

SBouchard01 commented:
I also noticed that the third exercise in the `PythonTutorial.ipynb` is missing the bold format on the instruction

hbrincon commented on desihigh/nofz.py:
```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+def generate_lss_nz(output_path = '../data/lss_catalogs_nz.pickle'):
```
Fixed

hbrincon commented on desihigh/universe.py:
```diff
@@ -0,0 +1,31 @@
+
+def time_to_redshift(model, interp_redshift, time):
+
+    from scipy.interpolate import InterpolatedUnivariateSpline
```
Fixed

hbrincon commented:
@SBouchard01 I've updated the doctrings to the numpy standard

SBouchard01 commented:
I just made a final pass on the notebooks & code in this PR:
- I cleaned up the docstrings & type hints in the `desihigh` package (also solved a few missing imports)
- I cleaned up the notebooks and removed cells outputs and code solutions in the cells

Last step before merging to minimal is to merge #135 into this PR (with the notebooks cleaned and the merge conflicts resolved)
I also noticed that the following branches are stale, @hbrincon can you confirm that we can delete them after this merge ?
- `mapping-the-universe`
- `minimal-python-tutorial`
- `minimal-modules`
- `distance`
- `python_tutorial`
- `short_intro`

And finally before making the final merge to the minimal branch, @hbrincon could you check one last time that everything looks good and works ? :)

SBouchard01 commented:
I've merged `minimal-bao-redshift` in this PR, and added some dependencies in the `pyproject.toml`.

I'm waiting for @hbrincon approval & final review to merge this into the `minimal` branch, after I've tested that the installation and that all the notebooks solutions work as expected !

hbrincon commented:
@SBouchard01 Everything looks good! I've added a missing comment in one of the notebook solutions, and I've updated one of the notebook links to point to the website curriculum page for now (all inter-notebook links can be updated again once we've updated the main branch and have binder working)
